packages/server/src/fhir/operations/ai.ts

```diff
@@ -79,75 +80,208 @@ type AIOperationParameters = {
-        id: tc.id,
-        type: tc.type,
-        function: {
-          name: tc.function.name,
-          arguments: JSON.parse(tc.function.arguments),
-        },
-      }));
-
-      parameters.push({
-        name: 'tool_calls',
-        valueString: JSON.stringify(toolCallsWithParsedArgs),
-      });
+    // Set SSE headers
+    res.setHeader('Content-Type', 'text/event-stream');
+    res.setHeader('Cache-Control', 'no-cache');
+    res.setHeader('Connection', 'keep-alive');
+    res.flushHeaders();
+
+    await streamAIToClient(messages, params.apiKey, params.model, tools, res);
+    res.end();
+
+    // Return undefined for streaming - response already sent
+    return undefined;
+  }
+
+  try {
+    const result = (await callAI(messages, params.apiKey, params.model, tools)) as {
+      content: string | null;
+      tool_calls: any[];
+    };
+    return buildParametersResponse(result);
+  } catch (error) {
+    return [badRequest('Failed to call OpenAI API: ' + (error as Error).message)];
+  }
+}
+
+/**
+ * Streams AI response from OpenAI directly to the client via SSE.
+ * This function bridges the OpenAI stream to the Express response without collecting.
+ * Note: Tool calls are not supported in streaming mode.
+ * @param messages - The conversation messages
+ * @param apiKey - OpenAI API key
+ * @param model - Model to use
+ * @param tools - Optional tools array (ignored in streaming mode)
+ * @param res - Express response to write SSE data to
+ */
+export async function streamAIToClient(
+  messages: any[],
+  apiKey: string,
+  model: string,
+  tools: any[] | undefined,
+  res: ExpressResponse
+): Promise<void> {
+  const response = (await callAI(messages, apiKey, model, tools, true)) as Response;
+  if (!response.body) {
+    throw new Error('No response body available for streaming');
+  }
+
+  // Stream OpenAI response directly to client using TextDecoderStream
+  const reader = response.body.pipeThrough(new TextDecoderStream()).getReader();
+
+  let buffer = '';
+
+  try {
+    while (true) {
+      const { done, value } = await reader.read();
+
+      if (done) {
+        res.write('data: [DONE]\n\n');
+        break;
+      }
+
+      buffer += value;
+      const lines = buffer.split('\n');
+      buffer = lines.pop() || '';
+
+      for (const line of lines) {
+        if (line.startsWith('data: ')) {
+          const data = line.slice(6).trim();
+
+          if (data === '[DONE]') {
+            continue;
+          }
+
+          try {
+            const parsed = JSON.parse(data);
+            const delta = parsed.choices[0]?.delta;
+
+            if (!delta?.content) {
+              continue;
+            }
+
+            // Forward content chunks to client
+            res.write(`data: ${JSON.stringify({ content: delta.content })}\n\n`);
+          } catch (e) {
+            // Skip malformed JSON
+            console.error('Error parsing SSE data:', e);
+          }
+        }
+      }
     }
+  } finally {
```

Comment:
I'm thinking any error thrown in this `try` block might put the stream into a weird state 🤔 

Might want to add a catch here where we can at least log the error like above and then call `res.end` to trigger the error handling logic on the client

What do you think?